src/core/core.scale.js

```diff
@@ -93,10 +83,129 @@ function getPixelForGridLine(scale, index, offsetGridLines) {
 	return lineValue;
 }
 
-function computeTextSize(context, tick, font) {
-	return helpers.isArray(tick) ?
-		helpers.longestText(context, font, tick) :
-		context.measureText(tick).width;
+function measureText(ctx, data, gc, string) {
```

Comment:
Alternative implementation to use existing code and help simplify `computeLabelSize`:
```js
var cache = {};
function measureText(ctx, font, text) {
	var fontString = font.string;
	var fontCache = cache[fontString] = cache[fontString] || {};
	text = helpers.isArray(text) ? text : [text];
	return helpers.longestText(ctx, fontString, text, fontCache);
}
```

Reply:
I like this implementation that keeps a separate cache for each font as it will improve performance to lookup. However, `helpers.longestText` cannot be used because it doesn't keep enough number of labels in the cache. So, I have rather modified the `computeLabelSizes` code to maintain a separate cache for each font and use `helpers.measureText`.